pageserver/src/bin/pageserver.rs

```diff
@@ -626,6 +626,7 @@ fn start_pageserver(
                     conf.id,
                     local_disk_storage,
                     metrics_ctx,
+                    cancel,
                 )
                 .instrument(info_span!("metrics_collection"))
                 .await?;
```

Comment:
Will this throw an error if cancellation happened? In the `tokio::select!` block above, we return `Ok(())` on cancellation. I didn't look closely but ISTM that we should do return the same thing in both cases.

Reply:
44111b3f86d62d458eca33c3eadc81de4711c2a5